src/native/minipal/thread.c

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <minipal/thread.h>
+
+#include <stdint.h>
+#include <string.h>
+
+#if defined(__linux__)
+#include <unistd.h>
+#include <sys/syscall.h>
+#elif defined(__FreeBSD__)
+#include <pthread_np.h>
+#elif defined(__NetBSD__)
+#include <lwp.h>
+#elif defined(__HAIKU__)
+#include <kernel/OS.h>
+#endif
+
+#ifdef PTHREAD_MAX_NAMELEN_NP
+#define MINIPAL_MAX_THREAD_NAME_SIZE (PTHREAD_MAX_NAMELEN_NP - 1)
+#elif defined(__APPLE__)
+#define MINIPAL_MAX_THREAD_NAME_SIZE 63
+#elif defined(__FreeBSD__)
+#define MINIPAL_MAX_THREAD_NAME_SIZE MAXCOMLEN
+#elif defined(__HAIKU__)
+#define MINIPAL_MAX_THREAD_NAME_SIZE (B_OS_NAME_LENGTH - 1)
+#else
+#define MINIPAL_MAX_THREAD_NAME_SIZE 15
+#endif
+
+size_t minipal_get_current_thread_id(void)
+{
+#ifdef __wasm
+    return 0;
+#else
+    static _Thread_local size_t tid = 0;
```

Comment:
The cache in `ep_rt_aot_current_thread_get_id` is unnecessary now that it is here.